lib/temporal/workflow/errors.rb

```diff
@@ -52,6 +52,17 @@ def self.generate_error(failure, default_exception_class = StandardError)
         end
       end
 
+      def self.generate_error_for_child_workflow_start(event)
+        if event.attributes.cause == :START_CHILD_WORKFLOW_EXECUTION_FAILED_CAUSE_WORKFLOW_ALREADY_EXISTS
+          Temporal::WorkflowExecutionAlreadyStartedFailure.new(
+            "The child workflow could not be started because a workflow with its id already exists: #{event.attributes.workflow_id}",
```

Comment:
You might want to mention the mutual exclusion policy.

Reply:
Like the run_id, it isn't available in this context / stack frame. I could use the start_child_workflow history event - it's available there. Do you think it's worth the factoring necessary to do so?